lightning/src/ln/peer_handler.rs

```diff
@@ -1535,10 +1538,18 @@ where
 	}
 
 	fn do_attempt_write_data(
-		&self, descriptor: &mut Descriptor, peer: &mut Peer, force_one_write: bool,
+		&self, descriptor: &mut Descriptor, peer: &mut Peer, mut force_one_write: bool,
 	) {
-		let mut have_written = false;
-		while !peer.awaiting_write_event {
+		if !self.peer_should_read(peer) {
```

Comment:
`peer_should_read` should be renamed IMO because it sounds like it returns whether the peer should read, when it actually returns whether we should read from the peer